src/form.js

```diff
@@ -446,23 +389,12 @@ define([
             this.fireChange(mug);
         },
         /**
-         * Walk merged data and control trees calling valueFunc for each mug
-         *
-         * If common nodes in merged tree branches are not in the same order
-         * then the order of nodes in the control tree will be preserved, and
-         * all nodes in the data tree that are not also in the control tree will
-         * be visited last.
```

Comment:
Got to the parser. It looks like the data tree order will be preserved, and the control tree order will be ignored/lost. I'm surprised that there are no tests that verify this comment. Please add a test that has data and control trees with mismatched order and verify that the control tree order is preserved after a parse/write cycle.

I'm not so worried about the last part of this comment "all nodes in the data tree that are not also in the control tree will be visited last." I think it's safe to omit that requirement if that's hard to implement.